integration/testutil.go

```diff
@@ -506,3 +506,64 @@ func genRandomByteData(size int64) []byte {
 	rand.Read(b)
 	return b
 }
+
+func rebootContainerd(t *testing.T, sh *shell.Shell, customContainerdConfig, customSnapshotterConfig string) *testutil.RemoteSnapshotMonitor {
+	var (
+		containerdRoot    = "/var/lib/containerd/"
+		containerdStatus  = "/run/containerd/"
+		snapshotterSocket = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+		snapshotterRoot   = "/var/lib/soci-snapshotter-grpc/"
+	)
+
+	// cleanup directories
+	testutil.KillMatchingProcess(sh, "containerd")
+	testutil.KillMatchingProcess(sh, "soci-snapshotter-grpc")
+	removeDirContents(sh, containerdRoot)
+	if isDirExists(sh, containerdStatus) {
+		removeDirContents(sh, containerdStatus)
+	}
+	if isFileExists(sh, snapshotterSocket) {
+		sh.X("rm", snapshotterSocket)
+	}
+	if snDir := filepath.Join(snapshotterRoot, "/snapshotter/snapshots"); isDirExists(sh, snDir) {
+		sh.X("find", snDir, "-maxdepth", "1", "-mindepth", "1", "-type", "d",
+			"-exec", "umount", "{}/fs", ";")
+	}
+	removeDirContents(sh, snapshotterRoot)
+
+	// run containerd and snapshotter
+	var m *testutil.RemoteSnapshotMonitor
+	containerdCmds := shell.C("containerd", "--log-level", "debug")
+	if customContainerdConfig != "" {
+		containerdCmds = addConfig(t, sh, customContainerdConfig, containerdCmds...)
+	}
+	sh.Gox(containerdCmds...)
+	snapshotterCmds := shell.C("/usr/local/bin/soci-snapshotter-grpc", "--log-level", "debug",
+		"--address", snapshotterSocket)
+	if customSnapshotterConfig != "" {
+		snapshotterCmds = addConfig(t, sh, customSnapshotterConfig, snapshotterCmds...)
+	}
+	outR, errR, err := sh.R(snapshotterCmds...)
+	if err != nil {
+		t.Fatalf("failed to create pipe: %v", err)
+	}
+	m = testutil.NewRemoteSnapshotMonitor(testutil.NewTestingReporter(t), outR, errR)
+
+	// make sure containerd and soci-snapshotter-grpc are up-and-running
+	sh.Retry(100, "ctr", "snapshots", "--snapshotter", "soci",
+		"prepare", "connectiontest-dummy-"+xid.New().String(), "")
+
+	return m
+}
+
+func removeDirContents(sh *shell.Shell, dir string) {
```

Comment:
Can't you `rm -fr {dir}` instead of using `find`?

Reply:
This is a direct move of https://github.com/awslabs/soci-snapshotter/blob/cc303960c39d34d53066f8908495a116b046a25e/integration/pull_test.go#L1029-L1088, since it's more a `testutil` than some logic for `pull_test.go`. I didn't change any code so it's easier to review (e.g., only relocation with no slight change in a long `+++`.)

I agree `rm -rf dir` is more straightforward than original `find` way. Will update this function.

Reply:
It seems `rm -rf dir` directly will cause some [device busy error](https://github.com/awslabs/soci-snapshotter/actions/runs/3886387536/jobs/6631429030#step:4:928). I think we can keep this part of code unchanged here since this PR is a refactor and other PRs can utilize the containerd/snapshotter config dedup from this PR early.

I'll make a separate PR to fix this (that will also easier to review). WDYT @kzys

Reply:
Interesting. How about commenting the reasons to use `find` then? I'm fine not using `rm -fr`.

Reply:
added comments. thanks